### PR TITLE
Added more machine learning quiz questions and explanations

### PR DESCRIPTION
**Pull request by HARSHIT-SAXENA3007**
Added multiple questions and explanations related to machine learning concepts, including overfitting, gradient descent, dimensionality reduction, evaluation metrics, reinforcement learning, confusion matrix, and neural network challenges.

## PR Checklist

This PR is ready for review and meets the requirements set out
in [Suggestion how to contribute](https://github.com/Ebazhanov/linkedin-skill-assessments-quizzes/blob/main/CONTRIBUTING.md)

### DOD

- [x] I have added new quiz{'s}
- [ ] I have added new reference link{'s}
- [ ] I have made small correction/improvements

### Changes / Instructions

_Add instructions to me, please type here, thanks_

### CODE REVIEW

**Review by Ebazhanov — CHANGES_REQUESTED** [review]
left some comments

**Comment by HARSHIT-SAXENA3007** [comment]
Just made the necessary changes. Kindly have a look.

Regards
